src/main/java/org/apache/mesos/curator/CuratorStateStore.java

```diff
@@ -373,6 +382,32 @@ public void clearProperty(final String key) throws StateStoreException {
         }
     }
 
+    @Override
+    public List<Protos.Resource> getExpectedResources() {
```

Comment:
This makes it more likely that implementers get the ResourceCleaner for free.  While not strictly necessary, I would say this is a point we can have an opinion about what a well formed StateStore should look like.  It should make it possible to not leak resources.  There are a lot of interfaces already.  I think in practice every StateStore implementation should want to provide this capability, so let's enforce it.